wordpress.org/public_html/wp-content/plugins/theme-directory/rest-api/class-theme-preview.php

```diff
@@ -7,18 +7,69 @@
 class Theme_Preview {
 
 	function __construct() {
-		register_rest_route( 'themes/v1', 'preview-blueprint/(?<slug>[^/]+)', array(
+		register_rest_route( 'themes/v1', 'preview-blueprint/(?<slug>[^/]+)(/(?<version>[^/]+))?', array(
 			'methods'             => WP_REST_Server::READABLE,
 			'callback'            => array( $this, 'preview' ),
+			'args'                => array(
+				'slug' => array(
+					'type'              => 'string',
+					'required'          => 'true',
+					'sanitize_callback' => 'sanitize_key',
+				),
+				'version' => array(
+					'type'              => 'string',
+					'required'          => 'false',
+					'sanitize_callback' => static function ( $value ) {
+						if ( ! is_string( $value ) ) {
+							return '';
+						}
+						return preg_replace( '/[^0-9a-zA-Z.-]/', '', $value );
+					},
+				),
+				// If the theme isn't (yet) published, this can be set to force loading the post.
+				'preview_id' => array(
+					'type'              => 'integer',
+					'required'          => false,
+					'sanitize_callback' => 'absint',
+				),
+				// Whether to enable WP_DEBUG. ie. for testing.
+				'debug' => array(
+					'type'              => 'boolean',
+					'required'          => false,
+					'default'           => false,
+					'sanitize_callback' => 'rest_sanitize_boolean',
+				),
+				// Whether to install the Theme Unit TestData.
+				'install_theme_testdata' => array(
+					'type'              => 'boolean',
+					'required'          => false,
+					'default'           => false,
+					'sanitize_callback' => 'rest_sanitize_boolean',
+				),
+				// Whether to install the Theme Previewer plugins.
+				'theme_previewer' => array(
+					'type'              => 'boolean',
+					'required'          => false,
+					'default'           => true,
+					'sanitize_callback' => 'rest_sanitize_boolean',
+				),
+			),
```

Comment:
I think these should be removed, and instead a new endpoint of `/theme-review-preview/SLUG/VERSION` be added instead. That can build off the rest of this PR changes but would give future stability for older trac ticket links.